mono/utils/mono-state.c

```diff
@@ -466,24 +463,53 @@ mono_native_state_add_epilogue (JsonWriter *writer)
 	mono_json_writer_object_end (writer);
 }
 
+void
+mono_native_state_init (JsonWriter *writer)
+{
+	mono_native_state_add_prologue (writer);
+}
+
+char *
+mono_native_state_emit (JsonWriter *writer)
+{
+	mono_native_state_add_epilogue (writer);
+	return writer->text->str;
+}
+
+char *
+mono_native_state_free (JsonWriter *writer, gboolean free_data)
+{
+	mono_native_state_add_epilogue (writer);
+	gchar *output = NULL;
+
+	// Make this interface work like the g_string free does
+	if (!free_data)
+		output = g_strdup (writer->text->str);
+
+	mono_json_writer_destroy (writer);
+	return output;
+}
+
 void
 mono_summarize_native_state_begin (void)
 {
 	mono_json_writer_init_static ();
-	mono_native_state_add_prologue (&writer);
+	mono_native_state_init (&writer);
 }
 
 char *
 mono_summarize_native_state_end (void)
 {
-	mono_native_state_add_epilogue (&writer);
-	return writer.text->str;
+	return mono_native_state_emit (&writer);
 }
 
 void
 mono_summarize_native_state_add_thread (MonoThreadSummary *thread, MonoContext *ctx)
 {
-	mono_native_state_add_thread (&writer, thread, ctx);
+	
+	static gboolean not_first_thread = FALSE;
```

Comment:
No need to say `= FALSE` here.

Reply:
No, but I think it makes it more readable in general to people who aren't familiar with C memory specifics. Every time I merge code with a static variable without an initializer, someone asks "are you sure this memory gets zeroed and it's not just garbage?"

You're 100% correct in machine semantics. I don't disagree there.

Reply:
People need to know this detail in C to read much code.
It is a widespread idiom.

Another thing to consider, is to give the variable a longer `mono_`-prefixed name and put it at global scope. That way I can easily ask the debugger to see its value even if I'm not in the function. There is a scoping tradeoff though, no best answer.